presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -134,7 +135,10 @@ private PlanNode getCostBasedJoin(JoinNode joinNode, Context context)
             }
         }
 
-        if (possibleJoinNodes.stream().anyMatch(result -> result.getCost().hasUnknownComponents()) || possibleJoinNodes.isEmpty()) {
+        PlanNodeStatsEstimate buildSideStats = context.getStatsProvider().getStats(joinNode.getRight());
+        boolean isBuildSideStatsUnknown = buildSideStats.getOutputRowCount() == 0 && buildSideStats.confidenceLevel() == LOW;
```

Comment:
What does rowcount == 0 add here? Wouldn't it make more sense to just check for low confidence level for the build side?

Reply:
This is specifically to handle the case where multiple filters are being pushed in and it results in an estimate of 0. Trying to handle the misleading case where a 0 estimate does not correlate to it be truly 0